unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/FriendsController.cs

```diff
@@ -386,20 +449,49 @@ public async UniTask<List<FriendRequest>> GetFriendRequestsAsync(int sentLimit,
         public void GetFriendsWithDirectMessages(string userNameOrId, int limit) =>
             apiBridge.GetFriendsWithDirectMessages(userNameOrId, limit, 0);
 
-        public FriendRequest GetAllocatedFriendRequest(string friendRequestId) =>
-            friendRequests.ContainsKey(friendRequestId) ? friendRequests[friendRequestId] : null;
+        public FriendRequest GetAllocatedFriendRequest(string friendRequestId)
+        {
+            if (useSocialApiBridge)
+            {
+                return incomingFriendRequestsById.ContainsKey(friendRequestId) ? incomingFriendRequestsById[friendRequestId] :
+                    outgoingFriendRequestsById.ContainsKey(friendRequestId) ? outgoingFriendRequestsById[friendRequestId] : null;
+            }
+
+            return friendRequests.ContainsKey(friendRequestId) ? friendRequests[friendRequestId] : null;
+        }
 
         public FriendRequest GetAllocatedFriendRequestByUser(string userId)
         {
             long max = long.MinValue;
             FriendRequest result = null;
 
-            foreach (var request in friendRequests.Values)
+            if (useSocialApiBridge)
+            {
+                foreach (var request in incomingFriendRequestsById.Values)
+                {
+                    if (request.From != userId && request.To != userId) continue;
+                    if (request.Timestamp <= max) continue;
+                    result = request;
```

Comment:
if you already had a match here, then you should avoid iterating the outgoing friend requests (missing `return`)